frappe/utils/__init__.py

```diff
@@ -75,6 +76,21 @@ def extract_email_id(email):
 		email_id = email_id.decode("utf-8", "ignore")
 	return email_id
 
+def validate_phone_number_with_isd(phone, throw=False):
+	if not phone:
+		return
+	try:
+		phone_number = ph.parse(phone)
+	except Exception as e:
+		if e.error_type == 1:
+			frappe.throw(frappe._("{0} is not a valid Phone Number.").format(frappe.bold(phone)), frappe.InvalidPhoneNumberError,
+				title=frappe._("Invalid Number"))
```

Comment:
```suggestion
			frappe.throw(
				_("Phone number {0} set in field {1} is not valid").format(frappe.bold(phone), fieldname), 
				frappe.InvalidPhoneNumberError,
				title=_("Invalid Phone Number")
			)
```
- Let's mention the field name in which the invalid number is set.
- Import `_` at the top.
- Fix other messages as well.
- If this function is not generic enough, let's keep the code in `base_document.py`